### PR TITLE
T277552 project jdata store as parquet

### PR DESCRIPTION
**Pull request by gmodena**
Project `instanceof` metadata in the model output.

This PR adds change to export metadata related to the "instance of" 
property of a q item. This information stored as an appended to column to the model output,
and propagates to HDFS and Hive `imagerec` datasets.

This PR also adds a spark job to upload and convert model output to `Parquet`. This has
been done to facilitate interop with spark, and handle schema migrations.

Closes: https://phabricator.wikimedia.org/T277552

### CODE REVIEW

File: algorithm.ipynb [hunk 1]
```diff
@@ -539,7 +539,7 @@
     "        else:\n",
     "            top_candidates.append(None)\n",
     "    data_small[wiki]['top_candidates']=top_candidates\n",
-    "    data_small[wiki][['item_id','page_id','page_title','top_candidates']].to_csv(output_dir+'/'+wiki+'_'+snapshot+'_wd_image_candidates.tsv',sep='\\t')"
+    "    data_small[wiki][['item_id','page_id','page_title','top_candidates', 'instanceof']].to_csv(output_dir+'/'+wiki+'_'+snapshot+'_wd_image_candidates.tsv',sep='\\t')"
```
Comment (gmodena) [inline algorithm.ipynb:542]: Hey @mirrys. I wanted to flag this change in the notebook (`algorithm.ipynb`).

File: etl/schema.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+        .add("top_candidates", StringType(), True)
+        .add("instance_of", StringType(), True)
+    )
+
+
+class RawDataset(CsvDataset):
+    schema = CsvDataset.schema.add("wiki_db", StringType(), True).add(
+        "snapshot", StringType(), True
+    )
+    recommendation_schema = "array<struct<image:string,note:string,rating:double>>"
```
Comment (gmodena) [inline etl/schema.py:20]: I'm not happy with this boilerplate to begin with, but we should (at least) rename recommendation_schema to `top_candidates_schema`  (or similar) to reflect its purpose.

Comment (clarakosi) [reply etl/schema.py:20]: We can address this in my patch if its okay with you

**Review by clarakosi — APPROVED** [review]
One comment above but if its okay with you then we can merge this

File: etl/schema.py (hunk 2 above)
Comment (gmodena) [reply etl/schema.py:20]: Sorry, I did miss this comment. Happy to have your refactor in your patch, or anywhere else outside of this PR.

**Pull request merged**